code/graphics/paths/nanovg/stb_truetype.h

```diff
@@ -968,640 +1254,1433 @@ typedef int stbtt__test_oversample_pow2[(STBTT_MAX_OVERSAMPLE & (STBTT_MAX_OVERS
-      // Binary search the right group.
-      while (low < high) {
-         stbtt_int32 mid = low + ((high-low) >> 1); // rounds down, so low <= mid < high
-         stbtt_uint32 start_char = ttULONG(data+index_map+16+mid*12);
-         stbtt_uint32 end_char = ttULONG(data+index_map+16+mid*12+4);
-         if ((stbtt_uint32) unicode_codepoint < start_char)
-            high = mid;
-         else if ((stbtt_uint32) unicode_codepoint > end_char)
-            low = mid+1;
-         else {
-            stbtt_uint32 start_glyph = ttULONG(data+index_map+16+mid*12+8);
-            if (format == 12)
-               return start_glyph + unicode_codepoint-start_char;
-            else // format == 13
-               return start_glyph;
-         }
-      }
-      return 0; // not found
-   }
-   // @TODO
-   STBTT_assert(0);
-   return 0;
+	stbtt_uint8 *data = info->data;
+	stbtt_uint32 index_map = info->index_map;
+
+	stbtt_uint16 format = ttUSHORT(data + index_map + 0);
+	if (format == 0) { // apple byte encoding
+		stbtt_int32 bytes = ttUSHORT(data + index_map + 2);
+		if (unicode_codepoint < bytes - 6)
+			return ttBYTE(data + index_map + 6 + unicode_codepoint);
+		return 0;
+	}
+	else if (format == 6) {
+		stbtt_uint32 first = ttUSHORT(data + index_map + 6);
+		stbtt_uint32 count = ttUSHORT(data + index_map + 8);
+		if ((stbtt_uint32)unicode_codepoint >= first && (stbtt_uint32)unicode_codepoint < first + count)
+			return ttUSHORT(data + index_map + 10 + (unicode_codepoint - first) * 2);
+		return 0;
+	}
+	else if (format == 2) {
+		STBTT_assert(0); // @TODO: high-byte mapping for japanese/chinese/korean
+		return 0;
+	}
+	else if (format == 4) { // standard mapping for windows fonts: binary search collection of ranges
+		stbtt_uint16 segcount = ttUSHORT(data + index_map + 6) >> 1;
+		stbtt_uint16 searchRange = ttUSHORT(data + index_map + 8) >> 1;
+		stbtt_uint16 entrySelector = ttUSHORT(data + index_map + 10);
+		stbtt_uint16 rangeShift = ttUSHORT(data + index_map + 12) >> 1;
+
+		// do a binary search of the segments
+		stbtt_uint32 endCount = index_map + 14;
+		stbtt_uint32 search = endCount;
+
+		if (unicode_codepoint > 0xffff)
+			return 0;
+
+		// they lie from endCount .. endCount + segCount
+		// but searchRange is the nearest power of two, so...
+		if (unicode_codepoint >= ttUSHORT(data + search + rangeShift * 2))
+			search += rangeShift * 2;
+
+		// now decrement to bias correctly to find smallest
+		search -= 2;
+		while (entrySelector) {
+			stbtt_uint16 end;
+			searchRange >>= 1;
+			end = ttUSHORT(data + search + searchRange * 2);
+			if (unicode_codepoint > end)
+				search += searchRange * 2;
+			--entrySelector;
+		}
+		search += 2;
+
+		{
+			stbtt_uint16 offset, start;
+			stbtt_uint16 item = (stbtt_uint16)((search - endCount) >> 1);
+
+			if (unicode_codepoint > ttUSHORT(data + endCount + 2 * item)) {
+				nvgOldCPrintf("WARNING: Probably font character is missing. Unicode decimal index is %d.\n", unicode_codepoint);
```

Comment:
The warning has a few grammar issues. It should probably be `WARNING: A font character is probably missing. Unicode decimal index is %d.`.